src/app/landing-page/landing-page.component.ts

```diff
@@ -0,0 +1,54 @@
+import { Component } from '@angular/core';
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { environment } from '../../environments/environment';
+import { Router } from '@angular/router';
+
+@Component({
+  selector: 'app-landing-page',
+  templateUrl: './landing-page.component.html',
+  styleUrls: ['./landing-page.component.scss']
+})
+export class LandingPageComponent {
+  emailValid: boolean = null;
+  captchaSiteKey: string = environment.captchaSiteKey;
+  captchaResponse: string = null;
+  unsuccessfulMessage: string;
+
+  constructor(private http: HttpClient,
+              private router: Router) {}
+
+  submitEmail(email: string) {
+    this.validateEmail(email);
+    if (this.emailValid && this.captchaResponse !== null) {
+      this.router.navigateByUrl('/success');
+      const payload = JSON.stringify({"text": email + " wants to join the Swiftfest Slack Channel!", "channel": "submissions"});
+      const headers = new HttpHeaders().append("Content-type", "application/x-www-form-urlencoded; charset=UTF-8");
+      this.http.request(
+        "POST",
+        environment.slackWebHookUrl,
+        {
+          "body": payload,
+          "headers": headers,
+          "responseType": "text"
+        }
+      ).subscribe();
+      this.unsuccessfulMessage = null;
```

Comment:
Why `null` instead of an empty string?

Reply:
It works the same way null or empty string for the condition. I have changed it to empty string